dataset/src/main/scala/frameless/TypedDataset.scala

```diff
@@ -5,12 +5,15 @@ import org.apache.spark.rdd.RDD
 import org.apache.spark.sql.catalyst.expressions.{Alias, CreateStruct, EqualTo}
 import org.apache.spark.sql.catalyst.plans.{Inner, LeftOuter}
 import org.apache.spark.sql.catalyst.plans.logical.{Join, Project}
-import org.apache.spark.sql.{FramelessInternals, Column, SQLContext, Dataset}
-import shapeless.ops.hlist.{Tupler, ToTraversable}
+import org.apache.spark.sql.{Column, Dataset, FramelessInternals, SQLContext}
+import shapeless.ops.hlist.{ToTraversable, Tupler}
 import shapeless._
 
 /** [[TypedDataset]] is a safer interface for working with `Dataset`.
   *
+  * NOTE: Prefer `TypedDataset.create` over `new TypedDataset` unless you
```

Comment:
Point taken. What about a `private` constructor, a public `TypedDataset.create` and another public `TypedDataset.unsafeCreate`? I'm not the scaladoc is enough of a guard, I personally just randomly use stuff without looking at the doc and expect it to work :P